core/src/main/scala/org/apache/spark/sql/execution/row/RowFormatRelation.scala

```diff
@@ -214,6 +216,48 @@ class RowFormatRelation(
     }
     s"CREATE $indexType INDEX $indexName ON $baseTable ($columns)"
   }
+
+  /** Base table of this relation. */
+  override def baseTable: Option[String] = tableOptions.get(StoreUtils.COLOCATE_WITH)
+
+  /** Name of this relation in the catalog. */
+  override def name: String = table
+
+  override def addDependent(dependent: DependentRelation,
+      catalog: SnappyStoreHiveCatalog): Boolean =
+    DependencyCatalog.addDependent(table, dependent.name)
+
+  override def removeDependent(dependent: DependentRelation,
+      catalog: SnappyStoreHiveCatalog): Boolean =
+    DependencyCatalog.removeDependent(table, dependent.name)
+
+
+  override def getDependents(
+      catalog: SnappyStoreHiveCatalog): Seq[String] =
+    DependencyCatalog.getDependents(table)
+
+  /**
+   * Recover/Re-create the dependent child relations. This callback
+   * is to recreate Dependent relations when the ParentRelation is
+   * being created.
+   */
+  override def recoverDependentsRelation(properties: Map[String, String]): Unit = {
+
+    val snappySession = sqlContext.sparkSession.asInstanceOf[SnappySession]
+    val sncCatalog = snappySession.sessionState.catalog
+
+    var dependentRelations: Array[String] = Array()
+    if (None != properties.get(ExternalStoreUtils.DEPENDENT_RELATIONS)) {
+      dependentRelations = properties(ExternalStoreUtils.DEPENDENT_RELATIONS).split(",")
+    }
+    dependentRelations.foreach(rel => {
+      val dr = sncCatalog.lookupRelation(sncCatalog.newQualifiedTableName(rel)) match {
+        case LogicalRelation(r: DependentRelation, _, _) => r
+      }
+      addDependent(dr, sncCatalog)
+    })
+
+  }
 }
 
 final class DefaultSource extends MutableRelationProvider {
```

Comment:
You would need to add this line:

val dependentRelations = parameters.remove(ExternalStoreUtils.DEPENDENT_RELATIONS)
( siimilar to ColumnFormatRelation. Please check )